src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -792,22 +794,25 @@ public ActivityTraceId(ReadOnlySpan<byte> idData, bool isUtf8Chars = false)
             _id1 = 0;
             _id2 = 0;
             _asHexString = null;
-            fixed (ulong* idPtr = &_id1)
+            if (isUtf8Chars)
             {
-                var outBuff = new Span<byte>(idPtr, sizeof(ulong) * 2);
-                if (isUtf8Chars)
-                {
-                    if (idData.Length != 32)
-                        throw new ArgumentOutOfRangeException(nameof(idData));
-                    ActivityTraceId.SetSpanFromHexUtf8Chars(outBuff, idData);
-                }
-                else
+                if (idData.Length != 32)
+                    throw new ArgumentOutOfRangeException(nameof(idData));
+                Utf8Parser.TryParse(idData.Slice(0, 16), out _id1, out _, 'x');
+                Utf8Parser.TryParse(idData.Slice(16, 16), out _id2, out _, 'x');
+                if (BitConverter.IsLittleEndian)
                 {
-                    if (idData.Length != 16)
-                        throw new ArgumentOutOfRangeException(nameof(idData));
-                    idData.CopyTo(outBuff);
+                    _id1 = BinaryPrimitives.ReverseEndianness(_id1);
```

Comment:
I am not 100% confident about reversing the endianness here, mainly because I am not sure why we  get the bytes as big endian which ends up requiring this reversal. I would have thought we would want: `if (!BitConverter.IsLittleEndian) { reverse }`, i.e. reverse only for BigEndian. Just highlighting it.

Reply:
I believe we want it the way it currently is.   Basically the standard wants the bytes written in byte-by-byte order, which is 'standard' for network stuff.  This happens to correspond to big-endian if you try to clump it as larger integers as we are doing here.

Reply:
We did not get any feedback on the API review on this API.   We can change it, but the sooner the better, as it is a breaking change.   The alternative is to pick one to use the span<byte> constructor and the other is a static (or make both methods that take strings (as Spans) to be static).    If there is precedent for resolving this I am happy to follow it.